modular_core/lib/src/di/bind_context.dart

```diff
@@ -25,42 +28,54 @@ abstract class BindContextImpl implements BindContext {
 
   @override
   @visibleForTesting
-  List<BindContract> getProcessBinds() =>
-      _binds.where((element) => !element.export).toList();
+  List<BindContract> getProcessBinds() => _binds;
```

Comment:
The `modular_core:2.1.0` that's already published in `pub.dev` is breaking our tests on the CI because of this line change.

Since the `getProcessBinds()` returns directly the `_binds` reference instead of a copy of the list all the other methods that use it will be actually be changing the `_binds` directly.

One example is that when you use `initModule` for tests the list of `getProcessBinds()` will be passed as parameter to the `BindContextImpl.changeBinds` and since `newBinds` is actually a memory reference to `_binds` the line 34 will cause `newBinds` to be empty on line 35

Reply:
Version 2.0.2 retracted.
Now the available version is 2.1.0 for the new modular version. Thanks for letting us know about this breakchange

Reply:
Use modular_test version 2.0.0

Reply:
We're using `modular_test:2.0.0` already.

The problem is the CI that when it installs all the libraries it gets the newest of them because `flutter_modular` defines a dependency of `modular_core: >= 2.0.3+1 <3.0.0`

My solution was to add the specific versions to our pubspec.yaml to avoid this version upgrade.

```
  flutter_modular: 5.0.3
  modular_core: 2.0.3+1
  modular_interfaces: 2.0.2
```

Reply:
Truth. The version range doesn't really work well with this.

Reply:
Retracted version 2.1.0 modular_core for 3.0.0.
I believe this resolves conflicts. Thanks again @FBisca .